integration-tests/src/test/java/io/quarkiverse/fury/it/FuryTest.java

```diff
@@ -24,4 +27,20 @@ public void testPojo() {
     public void testThirdPartyBar() {
         given().when().get("/fury/third_party_bar").then().statusCode(200).body(is("true"));
     }
+
+    @Test
+    public void testFuryBar() {
+        Bar bar = new Bar(1, "hello bar");
+        Fury fury = Fury.builder().requireClassRegistration(true).withName("Fury" + System.nanoTime()).build();
+        fury.register(Bar.class);
```

Comment:
We can't write code like this, the serialization and deserialization should have same class registration order. If you register Bar here, it allocate an ID x for Bar,  but the server allocate id x for ThirdPartyBar. This is why it fails. You could update like this:

```java
@FurySerialization(classId=200)
class Bar {
...
}
fury.register(Bar.class, 200);
```

Reply:
Yeah, would you like submit a pr?

Reply:
You mean a PR for `Fury`?